src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -746,7 +746,7 @@ private static boolean isReturnedVariable(AbstractFrame currentFrame, DetailAST
     private static boolean isAstInside(DetailAST tree, DetailAST ast) {
         boolean result = false;
 
-        if (tree.equals(ast)) {
+        if (isAstSimilar(tree, ast)) {
```

Comment:
With AST hidden away `equals(AST)` became `equals(Object)` and it doesn't work the same. AST equals only looked at type and text, so we can't implement a true equals that would include line/column because other tests would break.